modules/tinymce/src/themes/silver/main/ts/ui/toolbar/button/ToolbarButtons.ts

```diff
@@ -263,7 +301,16 @@ const renderSplitButton = (spec: Toolbar.ToolbarSplitButton, sharedBackstage: Ui
         comp.getSystem().getByDom(button).each((buttonComp) => Toggling.set(buttonComp, state));
       });
     },
-    isActive: () => SelectorFind.descendant(comp.element, 'span').exists((button) => comp.getSystem().getByDom(button).exists(Toggling.isOn))
+    isActive: () => SelectorFind.descendant(comp.element, 'span').exists((button) => comp.getSystem().getByDom(button).exists(Toggling.isOn)),
+    setText: (text: string) =>
+      AlloyTriggers.emitWith(comp.components()[0], updateMenuText, {
```

Comment:
Try to exist using direct child relationships for identifying descendants - it makes everything too coupled to the existing DOM structure. Is `comp.components()[0]` the same as the `descendant('span')` above? Could a similar approach be used instead?

Reply:
Agreed. Changed to use SelectorFind.descendant